stats/thresholds_parser_test.go

```diff
@@ -0,0 +1,144 @@
+/*
+ *
+ * k6 - a next-generation load testing tool
+ * Copyright (C) 2021 Load Impact
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package stats
+
+import (
+	"reflect"
+	"testing"
+)
+
+func Test_parseThresholdExpression(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name          string
+		input         string
+		wantCondition *thresholdExpression
+		wantErr       bool
+	}{
+		{"unknown expression's operator fails", "count!20", nil, true},
+		{"unknown expression's method fails", "foo>20", nil, true},
+		{"non numerical expression's value fails", "count>abc", nil, true},
+		{"valid threshold expression syntax", "count>20", &thresholdExpression{AggregationMethod: "count", Operator: ">", Value: 20}, false},
```

Comment:
nit: I in general do not like "keyless" struct literals, both because they are harder to read and because adding one filed means you need to change all of them, even if most are fine with their zero value.

I do think that in this case (especially with some single digit tests) it's fine and maybe even better for readability. If any other reviewer also wants them to be with keys, perhaps we should change it though :shrug: 

The other cases in this file are getting more and more in the territory of me wanting them to not be keyless.

Reply:
I was unsure about the established best practice. Pretty sure I've seen both styles throughout the codebase, but my memory could also be failing me. In general, and in the spirit of "explicit is better than implicit", I agree with you.

I'll add the struct keys 😸